File: 백준/Bronze/1085. 직사각형에서 탈출/직사각형에서 탈출.py
x, y, w, h = map(int, input().split())
center_x = w / 2
center_y = h / 2

if center_x - abs(x - center_x) < center_y - abs(y - center_y):
  print(int(center_x - abs(x - center_x)))
else:
  print(int(center_y - abs(y - center_y)))

File: 백준/Bronze/1085. 직사각형에서 탈출/직사각형에서 탈출.cc
#include <cmath>
#include <iostream>
using namespace std;

int main() {
  int x, y, w, h;
  double center_x, center_y;

  cin >> x >> y >> w >> h;
  center_x = double(w) / 2;
  center_y = double(h) / 2;

  if (center_x - abs(x - center_x) < center_y - abs(y - center_y))
    cout << int(center_x - abs(x - center_x)) << "\n";
  else
    cout << int(center_y - abs(y - center_y)) << "\n";
}